components/UserProvider.js

```diff
@@ -32,7 +32,7 @@ class UserProvider extends React.Component {
   };
 
   state = {
-    loadingLoggedInUser: true,
+    loadingLoggedInUser: Boolean(getFromLocalStorage(LOCAL_STORAGE_KEYS.ACCESS_TOKEN)),
```

Comment:
This is a small but global change regarding how we handle the "loading" status of the logged-in user. The goal here is to consider that the user will only be "loading" if there's a token in local storage. Otherwise, we can directly set it to false.

It was required here to provide a smoother experience when linking directly to the payment step, which depends on whether we will reload the user or not to fill in the profile step.